akd/src/auditor.rs

```diff
@@ -13,36 +13,58 @@ use winter_crypto::Hasher;
 
 use crate::{
     errors::{AkdError, AzksError},
-    proof_structs::AppendOnlyProof,
+    proof_structs::{AppendOnlyProof, SingleAppendOnlyProof},
     storage::memory::AsyncInMemoryDatabase,
     Azks,
 };
 
+// FIXME: Need to add error handling
 /// Verifies an audit proof, given start and end hashes for a merkle patricia tree.
 pub async fn audit_verify<H: Hasher + Send + Sync>(
-    start_hash: H::Digest,
-    end_hash: H::Digest,
+    hashes: Vec<H::Digest>,
     proof: AppendOnlyProof<H>,
 ) -> Result<(), AkdError> {
-    verify_append_only::<H>(proof, start_hash, end_hash).await
+    for i in 0..hashes.len() - 1 {
+        let start_hash = hashes[i];
+        let end_hash = hashes[i + 1];
+        verify_consecutive_append_only::<H>(
+            &proof.proofs[i],
+            start_hash,
+            end_hash,
+            proof.epochs[i] + 1,
+        )
+        .await?;
+    }
+    Ok(())
 }
 
 /// Helper for audit, verifies an append-only proof
-pub async fn verify_append_only<H: Hasher + Send + Sync>(
-    proof: AppendOnlyProof<H>,
+pub async fn verify_consecutive_append_only<H: Hasher + Send + Sync>(
+    proof: &SingleAppendOnlyProof<H>,
     start_hash: H::Digest,
     end_hash: H::Digest,
+    epoch: u64,
 ) -> Result<(), AkdError> {
-    let unchanged_nodes = proof.unchanged_nodes;
-    let inserted = proof.inserted;
+    // FIXME: Need to get rid of the clone here.
```

Comment:
nit: why? What's wrong with the clone?

Reply:
This is left for my cleanup PR but I suspect there may be a way to do this using references that doesn't cause the borrow-checker to complain. I at least want to _try_ to reduce clones since that operation is less efficient.